### PR TITLE
drivers: nrf_power_clock: Migrate to DTS.

### PR DESCRIPTION
**Pull request by pizi-nordic**
This PR migrates the nrf_power_clock driver to DTS.
This change is needed for proper device power management.

### CODE REVIEW

File: dts/arm/nordic/nrf51822.dtsi [hunk 1]
```diff
@@ -63,6 +63,7 @@
 			reg = <0x40000000 0x1000>;
 			interrupts = <0 1>;
 			status = "ok";
+			label = "CLOCK_0";
```
Comment (anangl) [inline dts/arm/nordic/nrf51822.dtsi]: I think we could use just "CLOCK" as the label, as we use "WDT", for example.

**Review by anangl — APPROVED** [review]
LGTM

File: dts/arm/nordic/nrf51822.dtsi (hunk 1 above)
Comment (pizi-nordic) [reply dts/arm/nordic/nrf51822.dtsi]: Community: What you prefer?

**Review by carlescufi — APPROVED** [review]
LGTM

File: dts/arm/nordic/nrf51822.dtsi (hunk 1 above)
Comment (carlescufi) [reply dts/arm/nordic/nrf51822.dtsi]: No opinion, but @anangl has the better overview so I would go with his judgement

Comment (ioannisg) [reply dts/arm/nordic/nrf51822.dtsi]: CLOCK is ok

**Review by ioannisg — COMMENTED** [review]
The changes look ok to me I am just concerned about the stuff we _still_ keep in the Kconfig; could these also be moved to DTS? E.g. the 32kHZ clock source - could it go out of the driver's Kconfig?

**Review by aescolar — COMMENTED** [review]
This will fail in simulation just like the previous similar change did.
You just need to apply the same kind of patch to the nrf52_bsim 
d5b691adba589d2e622be78614f775d17a9d31c4

**Comment by pizi-nordic** [comment]
Thank you for pointing that @aescolar!

File: dts/arm/nordic/nrf51822.dtsi (hunk 1 above)
Comment (pizi-nordic) [reply dts/arm/nordic/nrf51822.dtsi]: Changed.

**Comment by pizi-nordic** [comment]
> 
> 
> The changes look ok to me I am just concerned about the stuff we _still_ keep in the Kconfig; could these also be moved to DTS? E.g. the 32kHZ clock source - could it go out of the driver's Kconfig?

This might be done later. Now I have to port all Nordic stuff to DTS in order to move forward with power management.

File: subsys/power/device.c [hunk 2]
```diff
@@ -27,8 +27,8 @@ LOG_MODULE_DECLARE(power);
 #define NUM_CORE_DEVICES	4
 #define MAX_DEV_NAME_LEN	16
 static const char core_devices[NUM_CORE_DEVICES][MAX_DEV_NAME_LEN] = {
-	"clk_k32src",
-	"clk_m16src",
+	"CLOCK_0_32K",
+	"CLOCK_0_16M",
```
Comment (anangl) [inline subsys/power/device.c]: After renaming `"CLOCK_0"` to `"CLOCK"` these two entries should be updated, too.

Comment (pizi-nordic) [reply subsys/power/device.c]: Yes, Thank you for pointing that.

Comment (pizi-nordic) [reply subsys/power/device.c]: Fixed

**Review by cvinayak — APPROVED** [review]
LGTM

**Pull request merged**